javatests/arcs/android/storage/service/StorageServiceNgImplTest.kt

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.service
+
+import android.os.IBinder
+import android.os.Parcel
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.crdt.ParcelableCrdtType
+import arcs.android.storage.writeStoreOptions
+import arcs.core.data.CountType
+import arcs.core.storage.FixedDriverFactory
+import arcs.core.storage.StorageKey
+import arcs.core.storage.StorageKeyParser
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.UntypedProxyMessage
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.testutil.MockDriverProvider
+import arcs.core.storage.testutil.testWriteBackProvider
+import com.google.common.truth.Truth.assertThat
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class StorageServiceNgImplTest {
+
+  private val driverFactory = FixedDriverFactory(MockDriverProvider())
+  private val onProxyMessageCallback: suspend (StorageKey, UntypedProxyMessage) -> Unit =
+    { _: StorageKey, _: UntypedProxyMessage -> }
+  private val storageKey = RamDiskStorageKey("myCount")
+  private val messageCallback = object : IMessageCallback {
```

Comment:
Done! The `FakeStorageChannelCallback` makes the tests a lot easier to follow imo, which is great :) 

I've also created the `FakeMessageCallback` which is just empty at the moment. Since it's not important in these tests, I could have just create a dummy message callback (@jblebrun suggested in [this](https://github.com/PolymerLabs/arcs/pull/6603#discussion_r527083704) comment), but I've left it as is because I reckon `FakeMessageCallback` can be built upon and be useful in other tests.